src/sage/rings/function_field/drinfeld_modules/drinfeld_module.py

```diff
@@ -883,6 +891,209 @@ def action(self):
+            sage: phi.basic_j_invariant_parameters([2, 1])
+            Traceback (most recent call last):
+            ...
+            ValueError: indices must be distinct and sorted
+            sage: phi.basic_j_invariant_parameters('x')
+            Traceback (most recent call last):
+            ...
+            TypeError: indices must be None, a tuple or a list
+        """
+        r = self._gen.degree()
+        if coeff_indices is None:
+            if nonzero:
+                coeff_indices = [k for k, g in enumerate(
+                    self.coefficients(sparse=False)[1:-1], start=1) if g]
+            else:
+                coeff_indices = list(range(1, r))
+        # Check if coeff_indices is valid:
+        elif isinstance(coeff_indices, (tuple, list)):
+            coeff_indices = list(coeff_indices)
+            if not all(isinstance(k, (int, Integer)) for k in coeff_indices):
+                raise TypeError('coefficients indices must be integers')
+            if max(coeff_indices) >= r or min(coeff_indices) <= 0:
+                raise ValueError(f'indices must be > 0 and < {r}')
+            if not all(coeff_indices[i] < coeff_indices[i+1] for i in
+                       range(len(coeff_indices) - 1)):
+                raise ValueError('indices must be distinct and sorted')
+            if nonzero:
+                coeff_indices = [k for k in coeff_indices if self._gen[k]]
+        else:
+            raise TypeError('indices must be None, a tuple or a list')
+        # Create the equation and inequalities for the polyhedron:
+        q = self._Fq.order()
+        equation = [0]
+        inequalities = []
+        # Create the equation:
+        # d_1 (q - 1) + d_2 (q^2 - 1) + ... + d_{r-1} (q^{r-1} - 1)
+        #  = d_r (q^r - 1)
+        for idx, i in enumerate(coeff_indices):
+            equation.append(q**i - 1)
+
+            # Create inequalities of the form 0 <= delta_i
+            lower_bounds = [0] * (len(coeff_indices) + 2)
+            lower_bounds[idx + 1] = 1
+
+            # Create inequalities of the form
+            # delta_i <= (q^r - 1)/(q^{gcd(i,r)} - 1)
+            upper_bounds = [Integer((q**r - 1) / (q**(gcd(i, r)) - 1))]\
+                            + [0]*(len(coeff_indices) + 1)
+            upper_bounds[idx + 1] = -1
+
+            inequalities.extend((lower_bounds, upper_bounds))
+
+        equation.append(1 - q**r)
+
+        # Create the polyhedron defined by the equation and the
+        # inequalities.
+        polyhedron = Polyhedron(ieqs=inequalities, eqns=[equation])
+
+        # Compute its integral points
+        integral_points = polyhedron.integral_points()
+
+        return [(tuple(coeff_indices), tuple(p)) for p in integral_points
+                if gcd(p) == 1]
+
+    def basic_j_invariants(self):
+        r"""
+        Return a dictionary whose keys are the basic `j`-invariants
+        parameters and values are the corresponding `j`-invariant.
```

Comment:
Adding a bit of redundancy for the sake of clarity:

```suggestion
        Return a dictionary whose keys are all the basic `j`-invariants
        parameters and values are the corresponding `j`-invariant.
```